content/docs/querying/api.md

```diff
@@ -0,0 +1,341 @@
+---
+title: HTTP API
+sort_rank: 7
+---
+
+# HTTP API
+
+The current stable HTTP API is reachable under `/api/v1` on a Prometheus
+server. Any non-breaking additions will be added under that endpoint. There are
+no minor versions of the API.
+
+## Format overview
+
+The API response format is JSON. Every successful API request returns a `2xx`
+status code.
+
+Invalid requests that reach the API handlers return a JSON error object
+and the HTTP response code `422 Unprocessable Entity`
+([RFC4918](http://tools.ietf.org/html/rfc4918#page-78)). Other non-`2xx`
+codes may be returned from outer HTTP handling layers.
```

Comment:
Implementation detail.

Other non-`2xx` codes may be returned for errors occurring before reaching the application.

Mostly we want to return `400` for invalid JSON (not invalid JSON data) - right now we simply don't have any.